rest/data/creathost.go

```diff
@@ -3,19 +3,26 @@ package data
 import (
 	"net/http"
 
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/apimodels"
 	"github.com/evergreen-ci/evergreen/model/host"
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/rest"
-	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/units"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 )
 
-// DBCreateHostConnector supports `create.host` commands from the agent.
+// DBCreateHostConnector supports `host.create` commands from the agent.
 type DBCreateHostConnector struct{}
 
+func (*DBCreateHostConnector) CreateHostsForTask(taskID string, createHost apimodels.CreateHost) error {
+	j := units.NewTaskHostCreateJob(taskID, createHost)
+	return evergreen.GetEnvironment().RemoteQueue().Put(j)
+}
```

Comment:
why does this need to be in the connector? the routes themselves have access to a queue in their factories/constructors, so we might as well just put directly into it.

Reply:
Done.